quill-spark/src/main/scala/io/getquill/QuillSparkContext.scala

```diff
@@ -101,7 +103,7 @@ trait QuillSparkContext
       prepare(Nil)._2.foldLeft(string) {
         case (string, DatasetBinding(ds)) =>
           dsId += 1
-          val name = s"ds$dsId"
+          val name = s"ds${UUID.randomUUID().toString.replace("-","")}"
```

Comment:
I think a long identifier might be confusing to people if they look at the generates SQL. Wdyt of

```
s"ds$dsId(${Thread.currentThread().hashCode()})"
```

Reply:
Aren't there some cases where async operations in Futures, Monix tasks etc... can jump across threads? If so, that could still potentially lead to `Thread.currentThread().hashCode()` evaluating to the same thing in two different places at the same time, expecially under very high load.

Reply:
Also, if `Thread.currentThread().hashCode()` is running on a slave node, it may be possible that it returns the same thing as `Thread.currentThread().hashCode()` on a different node.

Reply:
Here's another alternative. Use `Thread.currentThread().hashCode()` and provide an argument e.g. `useStrongIsolation` that enables `UUID.randomUUID()` for anyone who's completely paranoid like me :)

Reply:
Let's try to find a simpler solution. What about we add a counter (`AtomicInteger`) to `QuillSparkContext` and use it to create a suffix? That way, we can guarantee that Quill always generates a new identifier.

Reply:
I need to double check how these aliases work across nodes ie whether the spark context’s local aliases can be shared across nodes. If yes, it’s a problem because one node can have an atomic-integer that overrides the atomic integer of another node. If not, this solution will work.

Reply:
hey @deusaquilus, any updates on this?

Reply:
Okay. Switching to AtomicInteger. I'd like `prepareString` to be `private[getquill]` so that I can override it in corner cases.